File: customToast/app/src/main/java/com/example/customtoast/MainActivity.java
package com.example.customtoast;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);
        textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                //Toast.makeText(MainActivity.this,"this is good",Toast.LENGTH_SHORT).show();
                LayoutInflater inflater = getLayoutInflater();
                View view= inflater.inflate(R.layout.cutom_toast_layout, null);

                TextView textView = view.findViewById(R.id.textView1);

                textView.setText("Ashish is good boy");

                Toast toast = new Toast(MainActivity.this);
                toast.setView(view);

                //toast.setGravity(Gravity.LEFT | Gravity.TOP , 150,150);
                toast.setGravity(Gravity.LEFT  , 150,150);

                toast.show();

            }
        });

    }
}

File: recyclerView/Newspaper_recycleview/app/src/main/java/com/example/newspaper_recycleview/MainActivity.kt
package com.example.newspaper_recycleview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.TokenWatcher
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() ,onclick{

    var list = ArrayList<String>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        data()
        var adapter = NewsAdapter(list,this);
        val recycleview = findViewById<RecyclerView>(R.id.recycleview);
        recycleview.adapter=adapter
        //recycleview.(adapter)
        recycleview.layoutManager= LinearLayoutManager(this)
    }
    fun data()
    {

        for(i in 0..12){
            list.add("He is no ${i} in good morning")
        }
    }

    override fun actiononclick(name:String) {
        Toast.makeText(this,"Ashish Ranjan",Toast.LENGTH_SHORT).show()
    }
}